core/src/main/java/org/jdbi/v3/core/internal/Throwables.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.internal;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+public class Throwables {
+    private Throwables() {
+        throw new UnsupportedOperationException("utility class");
+    }
+
+    /**
+     * throws t if it is an Error, returns t directly if it is unchecked, wraps t in a RuntimeException otherwise
+     */
+    public static RuntimeException uncheck(Throwable t) {
+        if (t instanceof Error) {
+            throw (Error) t;
+        }
+
+        if (t instanceof RuntimeException) {
+            return (RuntimeException) t;
```

Comment:
why not `throw` here (and below) as we do above?

Reply:
Concur

Reply:
I'd like the call site to always do the throw. We can throw it in here as well, but then the throw at the call site is entirely redundant.

Reply:
That's probably necessary for flow control. It's how Guava's `Throwables.propagate()` worked--the propagate method always threw the exception but had a `RuntimeException` return type so the caller could "throw" a result that was never returned, to satisfy the compiler

Reply:
[Guava](https://github.com/google/guava/blob/master/guava/src/com/google/common/base/Throwables.java#L239-L242) methods `throw` unconditionally.  They declare a return of `RuntimeException` or the like but never actually return.

Absent any particular reason to change it, let's copy that style.  No need to do anything dramatic like take the entire class.  I'll make the change unless you're online to do it now.

Reply:
We also need a release note.